docs/debugging.asciidoc

```diff
@@ -0,0 +1,30 @@
+[[debugging]]
+== Debugging the agent itself
+
+Hopefully the agent Just Works™ for you but depending on your situation the agent could need some tuning.
+
+Firstly, to know more about what's going on inside the agent, you can increase the amount of log messages it writes by setting the log level with the option `log_level = 0` -- `0` being the level of most messages, `DEBUG`.
+
+In your `config/elastic_apm.yml`:
+
+[source,yaml]
+----
+log_level: <%= Logger::DEBUG %>
+----
+
+[float]
+[[debugging-log-messages]]
+=== Log messages
+
+[float]
+[[debugging-errors-queue-full]]
+==== `Queue is full (256 items), skipping…`
+
+The agent has an internal queue that holds events after they are done, before they are safely serialized and sent to APM Server. To avoid using up all of your memory, the queue has a fixed size. Depending on your load and server setup, events may be added to the queue faster than they are consumed, hence the warning.
```

Comment:
```suggestion
The agent has an internal queue that holds events after they are done, and before they are safely serialized and sent to APM Server. To avoid using up all of your memory, this queue has a fixed size. Depending on your load and server setup, events may be added to the queue faster than they are consumed, hence the warning.
```